### PR TITLE
Add ability for registry to specify input (IC) file variable names for constituents

### LINKED ISSUE #255: Allow registry to specify possible constituent variable names in IC files
### What is the feature/what would you like to discuss?

Right now all variables that are stored in the CCPP constituents object can only be read in from a file in CAM-SIMA if the file contains a variable with the exact standard name, which currently requires a separate dictionary and script to convert files with alternative variable names into standard names, which is a maintenance burden and reduces model flexibility.

An alternative to this is to utilize the `ic_file_input_names` tag within the CAM-SIMA registry to provide a list of possible input variable names for a given constituent standard name, which can then be implemented in the auto-generated `physics_inputs.F90` file so that CAM-SIMA scans an input file for the various possible constituent variable input names, instead of just the standard name, similar to what can be done for non-constituent variables now.

### Is there anyone in particular you want to be part of this conversation?

@peverwhee @mwaxmonsky @cacraigucar 

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

Any Software Engineer can do this

### PR DESCRIPTION
**Pull request by nusbaume**
Add the ability for the registry to specify the possible variable names for constituents on an input or Initial Conditions (IC) file in CAM-SIMA.  This should eliminate the need for an IC file to have only standard names for each variable, as long as the variable has the possible names listed in the registry.

Also, the `inputnames_to_stdnames.py` script and associated `stdnames_to_inputnames_dictionary.xml` file have now been moved to a new `tools` directory, as they should no longer be required for running CAM-SIMA (but could still potentially be useful for other situations).

Finally, changes were made to remove the unused `config` variable from the registry generator, as pylint found that it wasn't actually being used for anything. 

Fixes #255

### CODE REVIEW

**Review by peverwhee — COMMENTED** [review]
one question and a couple of typos

File: src/data/write_init_files.py [hunk 1]
```diff
@@ -338,13 +340,23 @@ def gather_ccpp_req_vars(cap_database):
...
+                    #until runtime, but still need variable names in order
+                    #to read from a file if need be:
+                    req_vars[stdname] = cvar
+
+                    #Add variable to constituent set:
+                    constituent_vars.add(stdname)
+                else:
+                    # We need to work with the host model version of this variable
+                    missing = _find_and_add_host_variable(stdname, host_dict,
+                                                          const_dicts, req_vars)
```
[the remark this answers is not in the document]
Comment (nusbaume) [reply src/data/write_init_files.py]: Well I tried running all of my tests (including doing snapshot runs) without it and it made no difference, so I am going to go with "no".  `const_dicsts` has now been removed.

File: src/data/write_init_files.py [hunk 2]
```diff
@@ -1236,7 +1284,22 @@ def write_phys_check_subroutine(outfile, host_dict, host_vars, host_imports,
     outfile.write("if (constituent_idx > -1) then", 4)
     outfile.comment("The required variable is a constituent. Call check variable routine on the relevant index of the constituent array", 5)
     outfile.write("field_data_ptr => cam_advected_constituents_array()", 5)
-    outfile.write("call check_field(file, [ccpp_required_data(req_idx)], 'lev', timestep, field_data_ptr(:,:,constituent_idx), ccpp_required_data(req_idx), min_difference, min_relative_value, is_first)", 5)
+    outfile.blank_line()
+    outfile.comment("Check if constituent standard name in registered SIMA standard names list:", 5)
+    outfile.write("if(any(phys_var_stdnames == ccpp_required_data(req_idx))) then", 5)
+    outfile.comment("Find array index to extract coorect input names:", 6)
```
[the remark this answers is not in the document]
Comment (nusbaume) [reply src/data/write_init_files.py]: It was a harrowing journey, but I believe I have fixed it.

**Review by mwaxmonsky — APPROVED** [review]
Looks great @nusbaume!

Was hoping to at least find more spelling errors but it looks like @peverwhee beat me to the punch on them.  ☹️

**Review by peverwhee — APPROVED** [review]
looks great!

**Pull request merged**